edb/schema/inheriting.py

```diff
@@ -29,27 +30,47 @@
 from . import name as sn
 from . import objects as so
 from . import utils
+from edb.schema import schema as s_schema
+
+if TYPE_CHECKING:
+    from edb.schema import objtypes as s_objtypes
```

Comment:
`inheriting.py` is a generic module, you shouldn't be using concrete implementations for the annotations.

Reply:
I understand: the concept of inheritance is generic. Just to be 100% sure, you mean that I shouldn't import `objtypes` module because it contains concrete classes, right? This wasn't clear to me while I started annotating this module: the main reason is that the `scls` variable (so.InheritingObjectBase) is used calling methods that are not defined in this base class, like `get_default_base_name` and `inheritable_fields`, which are defined on higher level classes.

In such situations, can we add abstract methods on the base class, to clarify the meaning of the class and its "interface" / "protocol", and satisfy mypy at the same time?

For example, adding to _InheritingObjectBase_:
```py
    def inheritable_fields(self) -> Iterable[Any]:
        raise NotImplementedError

    @classmethod
    def get_default_base_name(self) -> str:
        raise NotImplementedError
```

This is how I would do it (actually, in such cases I use @abstactmethod decorator from abc); do you agree with such approach or do you apply something else?